### PR TITLE
GPC-161: Add github oidc

### PR DESCRIPTION
**Pull request by oskarwilliams378**

### CODE REVIEW

File: terraform/modules/data-share-service/github_iam.tf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  }
+}
+
+resource "aws_iam_role" "github_oidc_environment" {
+  name               = "${var.environment}-github-oidc-deploy"
+  assume_role_policy = data.aws_iam_policy_document.github_oidc_environment_assume.json
+}
+
+data "aws_iam_policy" "github_oidc_environment" {
+  name = "Admin"
```
Comment (ethanmills) [inline terraform/modules/data-share-service/github_iam.tf:32]: I know this isn't a change, but I find this very scary

File: terraform/modules/data-share-service/github_iam.tf [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+resource "aws_iam_role" "github_oidc_pull_request" {
+  count = var.environment == "dev" ? 1 : 0
+
+  name               = "github-oidc-pull-request"
+  assume_role_policy = data.aws_iam_policy_document.github_oidc_pull_request_assume.json
+}
+
+data "aws_iam_policy" "github_oidc_pull_request" {
+  name = "ReadOnlyAccess"
+}
```
Comment (ethanmills) [inline terraform/modules/data-share-service/github_iam.tf:74]: Oh this is nice - so we get read only access when running PR changes, then full write when deploying? Nice!

File: .github/workflows/build-and-push.yml [hunk 3]
```diff
@@ -53,8 +52,8 @@ jobs:
       - name: Configure AWS credentials
         uses: aws-actions/configure-aws-credentials@v1
         with:
-          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          role-to-assume: arn:aws:iam::776473272850:role/dev-github-oidc-deploy
```
Comment (ethanmills) [inline .github/workflows/build-and-push.yml:55]: We could probably lock this down further at some point to just a role that has e.g. ECR full access. Not now though 🙂

**Review by ethanmills — APPROVED** [review]
Presumably this may need two attempts at deploying?

File: terraform/modules/data-share-service/github_iam.tf (hunk 1 above)
Comment (oskarwilliams378) [reply terraform/modules/data-share-service/github_iam.tf:32]: I am happy for us to lock this down if you know what terraform needs

File: terraform/modules/data-share-service/github_iam.tf (hunk 2 above)
Comment (oskarwilliams378) [reply terraform/modules/data-share-service/github_iam.tf:74]: Yep!

**Pull request merged**